dht/token.cc

```diff
@@ -112,4 +112,13 @@ token token::get_random_token() {
     return {kind::key, dht::get_random_number<int64_t>()};
 }
 
+token token::from_sstring(const sstring& t) {
+    auto lp = boost::lexical_cast<long>(t);
```

Comment:
I wonder why this function uses `long` here and not `int64_t`. Probably because that's what Origin did with Java.
I wonder what happens if we run this on a 32bit machine and the token doesn't fit in 32 bits.

Reply:
I didn't want to change the logic to make it easier for reviewers. I will fix this and your other comment on this commit in a separate patch (but in this PR).